### PR TITLE
json: add document for json_storage_free

### PR DESCRIPTION
**Pull request by YangKeao**
Signed-off-by: YangKeao <yangkeao@chunibyo.icu>

### What is changed, added or deleted? (Required)

Add documents for a new function `json_storage_free`.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/10943
- Other reference link(s): https://github.com/pingcap/tidb/issues/38517

### CODE REVIEW

**Comment by YangKeao** [comment]
@xiongjiwei PTAL

**Review by xiongjiwei — APPROVED** [review]

File: functions-and-operators/json-functions.md [hunk 1]
```diff
@@ -59,6 +59,7 @@ TiDB 支持 MySQL 5.7 GA 版本发布的大多数 JSON 函数。
 | 函数                     | 功能描述 |
 | --------------------------------- | ----------- |
 | [JSON_PRETTY(json_doc)][json_pretty] |格式化 JSON 文档 |
+| [JSON_STORAGE_FREE(json_doc)][json_storage_free] | 返回因局部更新而空出的 JSON 值的字节数。由于 TiDB 不支持局部更新 JSON 值，因此函数对合法的 JSON 值总是返回 0 |
```
[the remark this answers is not in the document]
Comment (songrijie) [reply functions-and-operators/json-functions.md:62]: “由于 TiDB 不支持局部更新 JSON 值” 可以改成 “由于 TiDB 采用完全不同的存储结构”

[the remark this answers is not in the document]
Comment (songrijie) [reply functions-and-operators/json-functions.md:62]: “返回该 JSON 对象的存储空间中空闲的字节数。 由于 TiDB 采用与 MySQL 完全不同的存储结构，本函数对合法的 JSON 值总是返回 0 ， 主要做兼容用途。 ”

[the remark this answers is not in the document]
Comment (YangKeao) [reply functions-and-operators/json-functions.md]: Fixed :heart:

File: functions-and-operators/json-functions.md [hunk 2]
```diff
@@ -59,6 +59,7 @@ TiDB 支持 MySQL 5.7 GA 版本发布的大多数 JSON 函数。
 | 函数                     | 功能描述 |
 | --------------------------------- | ----------- |
 | [JSON_PRETTY(json_doc)][json_pretty] |格式化 JSON 文档 |
+| [JSON_STORAGE_FREE(json_doc)][json_storage_free] | 返回该 JSON 对象的存储空间中空闲的字节数。 由于 TiDB 采用与 MySQL 完全不同的存储结构，本函数对合法的 JSON 值总是返回 0 ， 主要做兼容用途 |
```
[the remark this answers is not in the document]
Comment (YangKeao) [reply functions-and-operators/json-functions.md]: Fixed!

**Review by qiancai — APPROVED** [review]

**Review by Oreoxmt — APPROVED** [review]
LGTM

File: functions-and-operators/json-functions.md [hunk 3]
```diff
@@ -59,6 +59,7 @@ TiDB 支持 MySQL 5.7 GA 版本发布的大多数 JSON 函数。
 | 函数                     | 功能描述 |
 | --------------------------------- | ----------- |
 | [JSON_PRETTY(json_doc)][json_pretty] |格式化 JSON 文档 |
+| [JSON_STORAGE_FREE(json_doc)][json_storage_free] | 返回该 JSON 对象的存储空间中空闲的字节数。由于 TiDB 采用与 MySQL 完全不同的存储结构，本函数对合法的 JSON 值总是返回 0，主要做兼容用途 |
```
Comment (Oreoxmt) [inline functions-and-operators/json-functions.md:62]:
```suggestion
| [JSON_STORAGE_FREE(json_doc)][json_storage_free] | 返回该 JSON 对象的存储空间中空闲的字节数。由于 TiDB 采用与 MySQL 完全不同的存储结构，本函数对合法的 JSON 值总是返回 0，主要用于兼容 MySQL 8.0 |
```

Comment (YangKeao) [reply functions-and-operators/json-functions.md]: Fixed.

**Pull request merged**